### ISSUE TITLE
Freeze when dragging in-game windows

### ISSUE THREAD

**Issue by xdMatthewbx** [JGRennison/OpenTTD-patches#170]
## Version of OpenTTD
Latest on AUR "openttd-jgrpp" 0.34.4-1
Also happens on latest commit of fork by embeddedt (self-compiled)
No issue present on latest vanilla commit

## Expected result
The program doesn't freeze

## Actual result
The program freezes (sometimes indefinitely, sometimes it will unfreeze after about 2 minutes)

## Steps to reproduce
1. Launch game
2. Drag any in game window
3. killall -9 openttd-jgrpp

## GDB BT (AUR)
```
#0  0x00005555558ece1f in  ()
#1  0x00005555558ecf59 in  ()
#2  0x00005555558ecf1d in  ()
#3  0x00005555558ecf1d in  ()
#4  0x00005555558ecf1d in  ()
#5  0x00005555558ecf9c in  ()
#6  0x00005555558ecf9c in  ()
#7  0x00005555558ecf9c in  ()
#8  0x00005555558ecf9c in  ()
#9  0x00005555558ecf9c in  ()
#10 0x00005555558ecf1d in  ()
#11 0x00005555558ecf1d in  ()
#12 0x00005555558ecf1d in  ()
#13 0x00005555558ecf9c in  ()
#14 0x00005555558ecf9c in  ()
#15 0x00005555558ecf9c in  ()
#16 0x00005555558ecf1d in  ()
#17 0x00005555558ecf9c in  ()
#18 0x00005555558ecf1d in  ()
#19 0x00005555558ecf9c in  ()
#20 0x00005555558ecf1d in  ()
#21 0x0000555555c7097b in HandleMouseEvents() ()
#22 0x0000555555c37516 in VideoDriver_SDL::PollEvent() ()
#23 0x0000555555c37f08 in VideoDriver_SDL::MainLoop() ()
#24 0x0000555555a0b249 in openttd_main(int, char**) ()
#25 0x00007ffff71a3002 in __libc_start_main () at /usr/lib/libc.so.6
#26 0x00005555557a50de in _start ()
```

## GDB BT (fork by embeddedt) (included because I have the symbols for this one, I understand that this is not the place to post issues for forks)
```
(gdb) bt
#0  0x00005555558edfe6 in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1764
#1  0x00005555558ee15c in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1790
#2  0x00005555558ee15c in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1790
#3  0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#4  0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#5  0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#6  0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#7  0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#8  0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#9  0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#10 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#11 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#12 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#13 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#14 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#15 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#16 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#17 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#18 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#19 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#20 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#21 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#22 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#23 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#24 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#25 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#26 0x00005555558ee0dd in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1803
#27 0x00005555558ee15c in AddDirtyBlocks() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/gfx.cpp:1790
#28 0x0000555555c6d0e9 in HandleWindowDragging () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/window.cpp:2247
#29 MouseLoop () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/window.cpp:2944
#30 HandleMouseEvents() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/window.cpp:3111
#31 0x0000555555c33ae5 in VideoDriver_SDL::PollEvent() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/video/sdl2_v.cpp:811
--Type <RET> for more, q to quit, c to continue without paging--c
#32 0x0000555555c33fe8 in VideoDriver_SDL::LoopOnce() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/video/sdl2_v.cpp:1058
#33 0x0000555555c349b0 in VideoDriver_SDL::MainLoop() () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/video/sdl2_v.cpp:1044
#34 0x0000555555a0d365 in openttd_main(int, char**) () at /home/xdmatthewbx/Downloads/OpenTTD-modded/src/openttd.cpp:994
#35 0x00007ffff54f6002 in __libc_start_main () at /usr/lib/libc.so.6
#36 0x00005555557ac22e in _start ()
````

**Comment by JGRennison** [comment]
Thanks for reporting this.

I've not been able to reproduce this so far. Is there any particular window/window layout or anything else that you've noticed which seems to trigger this?

**Comment by xdMatthewbx** [comment]
> Thanks for reporting this.
> 
> I've not been able to reproduce this so far. Is there any particular window/window layout or anything else that you've noticed which seems to trigger this?

I believe it has to do with my system. I am running the latest version of Arch Linux with KDE Plasma.
I have 4 monitors, which might effect this. I had one of them set to 4k, perhaps that was a factor (although I did try it on a non-4k monitor and it still did the same thing).
If you're referring to in game windows I initially encountered this when I opened & attempted to drag a map window while hosting a game, then discovered it also happened on the title screen directly after launching if I drag the initial prompt.
I have also tried using a config from a system which is not experiencing this issue. No luck.
